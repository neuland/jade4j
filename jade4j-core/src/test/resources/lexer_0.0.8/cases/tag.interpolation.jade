
- var tag = 'p'
- var foo = 'bar'

#{tag} value
#{tag}(foo='bar') value
#{foo ? 'a' : 'li'}(something) here

mixin item(icon)
  li
    if attributes.href
      a&attributes(attributes)
        img.icon(src=icon)
        block
    else
      span&attributes(attributes)
        img.icon(src=icon)
        block

ul
  +item('contact') Contact
  +item(href='/contact') Contact
