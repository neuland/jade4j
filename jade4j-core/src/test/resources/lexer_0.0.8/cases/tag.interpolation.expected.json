{"type":"newline","line":2}
{"type":"code","line":2,"val":"var tag = 'p'","escape":false,"buffer":false}
{"type":"newline","line":3}
{"type":"code","line":3,"val":"var foo = 'bar'","escape":false,"buffer":false}
{"type":"newline","line":5}
{"type":"interpolation","line":5,"val":"tag"}
{"type":"text","line":5,"val":"value"}
{"type":"newline","line":6}
{"type":"interpolation","line":6,"val":"tag"}
{"type":"attrs","line":6,"attrs":[{"name":"foo","val":"'bar'","escaped":true}]}
{"type":"text","line":6,"val":"value"}
{"type":"newline","line":7}
{"type":"interpolation","line":7,"val":"foo ? 'a' : 'li'"}
{"type":"attrs","line":7,"attrs":[{"name":"something","val":true,"escaped":true}]}
{"type":"text","line":7,"val":"here"}
{"type":"newline","line":9}
{"type":"mixin","line":9,"val":"item","args":"icon"}
{"type":"indent","line":10,"val":2}
{"type":"tag","line":10,"val":"li","selfClosing":false}
{"type":"indent","line":11,"val":4}
{"type":"code","line":11,"val":"if ( attributes.href)","isElse":false,"isIf":true,"requiresBlock":true}
{"type":"indent","line":12,"val":6}
{"type":"tag","line":12,"val":"a","selfClosing":false}
{"type":"&attributes","line":12,"val":"attributes"}
{"type":"indent","line":13,"val":8}
{"type":"tag","line":13,"val":"img","selfClosing":false}
{"type":"class","line":13,"val":"icon"}
{"type":"attrs","line":13,"attrs":[{"name":"src","val":"icon","escaped":true}]}
{"type":"newline","line":14}
{"type":"mixin-block","line":14}
{"type":"outdent","line":15}
{"type":"outdent","line":15}
{"type":"code","line":15,"val":"else","isElse":true,"isIf":false,"requiresBlock":true}
{"type":"indent","line":16,"val":6}
{"type":"tag","line":16,"val":"span","selfClosing":false}
{"type":"&attributes","line":16,"val":"attributes"}
{"type":"indent","line":17,"val":8}
{"type":"tag","line":17,"val":"img","selfClosing":false}
{"type":"class","line":17,"val":"icon"}
{"type":"attrs","line":17,"attrs":[{"name":"src","val":"icon","escaped":true}]}
{"type":"newline","line":18}
{"type":"mixin-block","line":18}
{"type":"outdent","line":20}
{"type":"outdent","line":20}
{"type":"outdent","line":20}
{"type":"outdent","line":20}
{"type":"tag","line":20,"val":"ul","selfClosing":false}
{"type":"indent","line":21,"val":2}
{"type":"call","line":21,"val":"item","args":"'contact'"}
{"type":"text","line":21,"val":"Contact"}
{"type":"newline","line":22}
{"type":"call","line":22,"val":"item"}
{"type":"attrs","line":22,"attrs":[{"name":"href","val":"'/contact'","escaped":true}]}
{"type":"text","line":22,"val":"Contact"}
{"type":"outdent","line":23}
{"type":"eos","line":23}